src/model.js

```diff
@@ -0,0 +1,242 @@
+
+const proj = proj4("GOOGLE", "WGS84");
+
+function Model() {}
+
+// each model should have a getData() function to fetch the geo data
+// and format it into a geojson
+Model.prototype.getData = function (req, callback) {
+  if (req.query && req.query.returnCountOnly) {
+    return callback(null, {
+      type: "FeatureCollection",
+      count: 2001,
+      features: [],
+    });
+  }
+  const queries = buildQueries(req.query);
+  const featureCollection = {
+    type: "FeatureCollection",
+    features: [],
+  };
+
+  async.each(queries, search, finish);
+
+  function search(query, cb) {
+    // If this service is added to an ArcGIS Online Map, usually it makes multiple
+    // calls (4 quadrants). When you make 4 calls to the Yelp API at the same
+    // time, you get a Yelp error, so for now set a random pause 0.5-2 seconds.
+    // TODO: Remove this random pause code, and change it so that if we get a
+    // "TOO_MANY_REQUESTS_PER_SECOND" error, retry after a random pause.
+
+    const randomPauseTime = Math.random() * (2500 - 500) + 500;
+
+    setTimeout(() => {
+      searchYelp(query, function (err, features) {
+        if (err) return cb(err);
+        featureCollection.features = featureCollection.features.concat(
+          features
+        );
+        cb();
+      });
+    }, randomPauseTime);
+  }
+
+  function finish(err) {
+    // Service metadata
+    featureCollection.metadata = {
+      name: "Yelp",
+      description: `Generated from the Yelp API.`,
+    };
+
+    if (err) {
+      callback(err);
+    } else {
+      callback(null, featureCollection);
+    }
+  }
+};
+
+// Wrap the call to Yelp, this will make testing easier and decouple us from the specific client lib
+function searchYelp(query, callback) {
+  client.search(query).then(
+    function (rawResponse) {
+      // console.log("rawResponse", rawResponse);
+      const features = translate(rawResponse.jsonBody, query);
+      console.log("Search returned", features.length, "results");
+      callback(null, features);
+    },
+    function (err) {
+      // console.log("err", err);
+      callback(err);
+    }
+  );
+}
+
+function buildQueries(options) {
+  let queries;
+  if (options.geometry) {
+    queries = [buildQuery(options, options.geometry)];
+  } else {
+    queries = [buildQuery(options)];
+  }
+  return queries;
+}
+
+// Translate a request from the GeoServices API into something Yelp will understand
+function buildQuery(options, geometry) {
+  // we don't want to modify the passed in options object because Koop will use that later
+  // also for some reason the Yelp lib chokes when there is a passed in callback so omit it
+  const query = _.omit(_.cloneDeep(options), "callback");
+  if (geometry || options.geometry) {
+    const bbox = geometry || options.geometry;
+    [query.longitude, query.latitude] = getCenter(bbox);
+    query.radius = getRadius(bbox);
+  } else if (!options.location) {
+    query.location = "St. Louis, MO";
```

Comment:
Why `St. Louis, MO`?

Reply:
Why not? :)

Reply:
Does the query require a geometry/location so it needs a default value?

Reply:
yes